src/reactapp/src/components/checkoutAgreements/CheckoutAgreements.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+import Card from '../common/Card';
+import Header from '../common/Header';
+import { __ } from '../../i18n';
+import useCartContext from '../../hook/useCartContext';
+import Checkbox from '../common/Form/Checkbox';
+
+function CheckoutAgreements() {
+  const {
+    getCheckoutAgreements,
+    checkoutAgreements,
+    changeCheckoutAgreement,
+  } = useCartContext();
+
+  useEffect(() => {
+    getCheckoutAgreements();
+  }, [getCheckoutAgreements]);
+
+  if (!checkoutAgreements || checkoutAgreements.length < 1) {
+    return null;
+  }
+
+  return (
+    <div>
+      <Card>
+        <Header>{__('Checkout Agreements')}</Header>
+        {checkoutAgreements.map(agreement => (
+          <Checkbox
+            label={agreement.checkbox_text}
+            name={`agreement-${agreement.agreement_id}`}
+            required
+            isChecked={agreement.checked}
+            onChange={e => {
```

Comment:
Is there any reason we are keeping the agreements in the global `CartContext` and updating the value there? I think this state should be a part of formik form. For this, we need to define a new formSection just like login section, billing address section and deal the logic there. We need to create a new component like `CheckoutAgreementsFormikProvider` and add form states, validation there.

I cannot see the validation of agreements in this commit. We need to do that while placing the order? if any of the agreements not signed, then throw an error stating it. For all this, a wrapper component is necessary